### PR TITLE
[Improvement][Docker/K8s] Support more configurations and service accesses, reduce the duplication configurations, improve image for python and the compatibility of python home

### LINKED ISSUE #5431: [Bug][K8s] Master and worker cannot get the right address with custom DNS in 1.3.6
**For better global communication, Please describe it in English. If you feel the description in English is not clear, then you can append description in Chinese(just for Mandarin(CN)), thx! **
**Describe the bug**
(I found a bug when I test v1.3.6, which did not happen in v1.3.5.I need to used host network dns policy on k8s, so I add config in templates/statefulset-dolphinscheduler-worker.yaml)
我在测试刚发布的v1.3.6的时候，发现一个关于worker的命名的bug，在之前的1.3.5中并未出现。
起因：因为spark的一些特性，我需要在k8s的worker上使用本地网络，因此我在templates/statefulset-dolphinscheduler-worker.yaml添加了：
      hostNetwork: true
      dnsPolicy: ClusterFirstWithHostNet
于是我发现1.3.6在worker中命名规则与1.3.5并不一致，可以看到1.3.5中：
(but I found there is a different behavior between 1.3.5 and 1.3.6, we can see in v1.3.5:)
image
而在1.3.6中:
(but in v1.3.6:)
image
可以看出1.3.6里worker的hostname把后面的.local截掉了。导致工作流运作起来的时候，master找不到对应的worker主机。
(we can see v1.3.6 the hostname of workers is cut, which makes the workflow doesn't work. because the Master can't find the correct worker dns address.)
image

**Expected behavior**
roll back to v1.3.5，make full hostname of the worker。

**Screenshots**

**Which version of Dolphin Scheduler:**
 -[v1.3.6]

**Additional context**

**Requirement or improvement**
- Please describe about your requirements or improvement suggestions.

### LINKED ISSUE #5458: [Improvement][install] update the yarn application url property value one-click deployment
**Describe the question**
update the yarn application url property value one-click deployment

```
sed -i ${txt} "s#yarn.application.status.address.*#yarn.application.status.address=http://${singleYarnIp}:8088/ws/v1/cluster/apps/%s#g" conf/common.properties

```

to

```
sed -i ${txt} "s#yarn.application.status.address.*#yarn.application.status.address=http://${singleYarnIp}:%s/ws/v1/cluster/apps/%s#g" conf/common.properties

```

**Which version of DolphinScheduler:**
 -[dev]

### PR DESCRIPTION
**Pull request by chengshiwen**
## Purpose of the pull request

*[Improvement][Docker/K8s] Support more configurations and service accesses, reduce the duplication configurations, improve image for python and the compatibility of python home*

This pr is just a synchronization from #5101, #5158, #5195 and #5224 in `1.3.6-prepare`

Close issue: close #5458, close #5431

Related issue: #5405

## Brief change log

4141a2bda [Improvement][Docker] Support more configurations and improve image for python
54bc84d73 [Improvement][K8s] Support more configurations, more service accesses, and reduce the duplication configurations
98567b7bc [Improvement][Python] Improve the compatibility of python home
83af56d93 [Improvement][Install] Fix install config and install.sh
cc92e9a7f [Improvement][Install] Fix workflow ut
fd70f51b8 [Improvement][Docker] Optimize docker-swarm/check script
2fdf01f74 [Improvement][DB] Update default username and password of database
cf23fc5e2 [Docker/K8s] Specify version for all images
9ff8869ba [Improvement][Docker] Optimize PS1 and WORKDIR
f810f1cdb [Improvement][SQL] Reduce heap size to 64m in create-dolphinscheduler.sh and upgrade-dolphinscheduler.sh
4508756e4 [Fix-5431][K8s] Fix master and worker cannot get the right address with custom DNS

## Verify this pull request

This change added tests and can be verified as follows:

  - *Manually verified the change by testing locally.*

### CODE REVIEW

**Comment by chengshiwen** [comment]
@dailidong @xingchun-chen 
Done! Please review when you have time, thanks so much!

File: script/create-dolphinscheduler.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ export JAVA_HOME=$JAVA_HOME
 export DOLPHINSCHEDULER_CONF_DIR=$DOLPHINSCHEDULER_HOME/conf
 export DOLPHINSCHEDULER_LIB_JARS=$DOLPHINSCHEDULER_HOME/lib/*
 
-export DOLPHINSCHEDULER_OPTS="-server -Xmx1g -Xms1g -Xss512k -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled -XX:LargePageSizeInBytes=128m -XX:+UseFastAccessorMethods -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=70"
+export DOLPHINSCHEDULER_OPTS="-server -Xms64m -Xmx64m -Xss512k -XX:+DisableExplicitGC -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled -XX:LargePageSizeInBytes=64m -XX:+UseFastAccessorMethods -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=70"
```
Comment (davidzollo) [inline script/create-dolphinscheduler.sh:29]: I want to ask a question:  do 64m will be OK？

**Review by davidzollo — APPROVED** [review]
+1
great job

File: script/create-dolphinscheduler.sh (hunk 1 above)
Comment (chengshiwen) [reply script/create-dolphinscheduler.sh:29]: it's ok

**Pull request merged**